examples/layout/advanced/mpl_canvas_size.enaml

```diff
@@ -0,0 +1,76 @@
+#------------------------------------------------------------------------------
+# Copyright (c) 2013, Nucleic Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+#------------------------------------------------------------------------------
+""" An example of how to override tolkit-supplied size hints on your widget.
+
+
+The toolkit-supplied size hint for MPLCanvas is 480 x 640 pixels, which
+prevents you from sizing the canvas smaller. This can be a problem when you are
+presenting a large grid of Matplotlib figures.
+
+This example demonstrates how to override the tookit-supplied size hint for
+MPLCanvas. By setting resist_width, resist_height, hug_height, and hug_width to
+'ignore', you can resize the figure almost freely in any direction.
+
+This serves to demonstrate how the resist, hug and limit constraints interact
+to control the size of a widget.
+
+<< autodoc-me >>
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from enaml.widgets.api import (Form, Label, MPLCanvas, ObjectCombo, VGroup,
+                               Window)
+
+
+def make_figure():
+    figure, axes = plt.subplots()
+    x = np.arange(1000) / 1000
+    y = np.sin(2 * np.pi * 5 * x)
+    axes.plot(x, y)
+
+    # Call the `tight_layout` method anytime the user resizes the main window.
+    # This ensures that the axes titles and axes ticks are adjusted so they
+    # remain visible even at small sizes..
+    figure.canvas.mpl_connect('resize_event', lambda e: figure.tight_layout())
+    return figure
+
+
+enamldef Main(Window):
+
+    Form:
+        Label:
```

Comment:
May be worth detailing what resist_, hug_, limit_ impact each. So that people don't need to go find it in the source for themselves which is not exactly trivial.

Reply:
Done.